src/Plugin/GraphQL/DataProducer/Entity/EntityUrl.php

```diff
@@ -28,13 +33,16 @@ class EntityUrl extends DataProducerPluginBase {
    * Resolver.
    *
    * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to create a canonical URL for.
+   * @param null|array $options
+   *   The options to provide to the URL generator.
    *
    * @return \Drupal\Core\Url
    *
    * @throws \Drupal\Core\Entity\EntityMalformedException
    */
-  public function resolve(EntityInterface $entity) {
-    return $entity->toUrl();
+  public function resolve(EntityInterface $entity, ?array $options) {
+    return $entity->toUrl('canonical', $options ?? []);
```

Comment:
is it a good idea to hard code 'canonical'? I already see the next pull request coming wanting to have a parameter for the link relationship as well. I think we need to make that a parameter as well and default to 'canonical'.